include/measurement_kit/common/settings.hpp

```diff
@@ -6,10 +6,46 @@
 #define MEASUREMENT_KIT_COMMON_SETTINGS_HPP
 
 #include <map>
+#include <sstream>
+#include <string>
+#include <measurement_kit/common/error.hpp>
 
 namespace mk {
 
-typedef std::map<std::string, std::string> Settings;
+class Item {
+ public:
+    Item() {}
+
+    template <typename T>
+    Item (T t){
+        std::stringstream ss;
+        ss << t;
+        _store = ss.str();
+    }
+
+    template <typename T>
+    T as () const {
+        std::stringstream ss;
+        ss.str(_store);
+        T t;
+        ss >> t;
+        if (!ss.eof()) {
```

Comment:
Could you add a comment above this line explaining the rationale of checking both `eof()` and `fail()`?